### PR TITLE
Initial commit for accordian

### PR DESCRIPTION
**Pull request by DipshiGoyal**
@vinodloha ,as disccussed I have added the functionality of accordian. Please review.
PFA
[accordian (2).zip](https://github.com/pagesource/atomic-react-components/files/3106594/accordian.2.zip)

### CODE REVIEW

File: lib/components/molecules/Accordian/Accordian.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+// @flow
```
Comment (amrit07) [inline lib/components/molecules/Accordian/Accordian.js:1]: Accordian generally works in group; current implementation doesn't handle this. see https://www.w3.org/TR/wai-aria-practices/examples/accordion/accordion.html for an example; you can also explore deatil and summary html5 elements for the same.

File: lib/components/molecules/Accordian/Accordian.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            Body Content
+          </AccordianBody>
+        ) : null}
+      </React.Fragment>
+    );
+  }
+}
+
+const AccordianHeader = ({ className, onClick, children }: Props): Node => (
+  <React.Fragment>
```
Comment (amrit07) [inline lib/components/molecules/Accordian/Accordian.js]: Redundant usage of Fragment. below as well

File: lib/components/molecules/Accordian/Accordian.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ * Accordian
+ *
+ */
+import { PureComponent } from 'react';
+import styled from 'styled-components';
+import type { Node } from 'react';
+import styles from './Accordian.style';
+import type { Props, State } from './types';
+
+class Accordian extends PureComponent<Props, State> {
```
Comment (vinodloha) [inline lib/components/molecules/Accordian/Accordian.js:16]: @DipshiGoyal i would expect you to built this component like this https://codepen.io/amitmojumder/pen/xOKjPW

**Review by vinodloha — CHANGES_REQUESTED** [review]
Please fix review comments, let me know if you need more info

File: lib/components/molecules/Accordian/Accordian.js (hunk 3 above)
Comment (DipshiGoyal) [reply lib/components/molecules/Accordian/Accordian.js:16]: The suggestion has been incorporated.

File: lib/components/molecules/Accordian/tests/Accordian.test.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { shallow } from 'enzyme';
+// import renderer from 'react-test-renderer';
+
+import Accordian from '../index';
+
+describe('<Accordian />', () => {
```
Comment (vinodloha) [inline lib/components/molecules/Accordian/tests/Accordian.test.js:4]: The test needs to be much better, you need to test the expend collapse functionality

File: lib/components/molecules/Accordian/Accordian.style.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import styled from 'styled-components';
+
+const AccordianWrapper = styled.div`
+  display: block;
+  width: 50%;
+  margin: 10px 0;
+`;
+
+const AccordianHeader = styled.div`
```
Comment (vinodloha) [inline lib/components/molecules/Accordian/Accordian.style.js]: This is not correct style of applying CSS, you can follow same BEM style in styled components too.

File: lib/components/molecules/Accordian/Accordian.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    const doesShow = this.state.isOpen;
+    this.setState({ isOpen: !doesShow });
+  };
+
+  render() {
+    const { accordianHeader, accordianBody, className } = this.props;
+
+    return (
+      <AccordianWrapper>
+        <AccordianHeader
```
Comment (vinodloha) [inline lib/components/molecules/Accordian/Accordian.js]: AccordianHeader will not be DIV but a summary tag, you should built it as separate component.  Same applies to Detail too. You can export them same as accordion and people can loop over them and build accordion groups.

**Comment by vinodloha** [comment]
@DipshiGoyal lets connect today, i feel there is some gap in our understanding for this component.

**Pull request closed without merging**